src/fia/fia/fia_state.rs

```diff
@@ -0,0 +1,131 @@
+use super::*;
+use crate::prelude::*;
+
+pub(super) struct FiaState<ID, Path, Product>
+where
+    ID: Hash,
+    Path: HasDerivationPath,
+    Product: HasHDPublicKey,
+{
+    phantom_id: PhantomData<ID>,
+    phantom_path: PhantomData<Path>,
+    phantom_product: PhantomData<Product>,
+    /// If this FIA is used to derive public keys we cannot skip any
+    /// factor source, we need to derive ALL keys, so this variable will
+    /// be `false`. However, for transaction signing, this variable will
+    /// be `true`, since we **might** be able to skip *some* factor sources
+    /// and have valid signed transactions with e.g. just override factors.
+    pub(super) supports_skipping_of_factor_sources: bool,
+
+    /// Factor Sources Left to use. When this is Vec empty, we are done.
+    pub(super) factor_sources: Vec<FactorSource>,
+}
+
+impl<ID, Path, Product> FiaState<ID, Path, Product>
+where
+    ID: Hash,
+    Path: HasDerivationPath,
+    Product: HasHDPublicKey,
+{
+    fn invalid_if_skipped(&self, factor_sources: &[FactorSource]) -> Vec<ID> {
```

Comment:
will return which Transactions would fail if a user would skip using the factor sources. 

Note that `factor_sources` will have a **single** value for all drivers except `DeviceFactorSource` kind. 

The idea here is that Using DeviceFactorSource should allow for hosts to load multiple mnemonics from secure storage in on go - thus prompting for biometrics only once. @micbakos-rdx if this does not work on Android, I remove support for that. But I think doable? Would be greater UX to prompt for biometrics ONCE instead for every mnemonic.